[ECS] skip not working properly

In my easy-coding-standard.neon I have this:

```
parameters:
    skip:
        PHP_CodeSniffer\Standards\PSR2\Sniffs\ControlStructures\ControlStructureSpacingSniff.SpacingAfterOpenBrace:
        PHP_CodeSniffer\Standards\Squiz\Sniffs\WhiteSpace\ControlStructureSpacingSniff.SpacingAfterOpenBrace:
```

(There is a bug in these two sniffs so I can't use them for now.)

The problem is that when I run ECS it throws some false-positive errors:

```
./vendor/bin/ecs check src --fix
 1/1 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

                                                                                                                        
 [OK] No errors found. Great job - your code is shiny in style!                                                         
                                                                                                                        

                                                                                                                        
 [ERROR] Skipped checker                                                                                                
         "PHP_CodeSniffer\Standards\PSR2\Sniffs\ControlStructures\ControlStructureSpacingSniff.SpacingAfterOpenBrace"   
         were not found. You can remove them from "parameters: > skip:" section in your config.                         
                                                                                                                        

                                                                                                                        
 [ERROR] Skipped checker                                                                                                
         "PHP_CodeSniffer\Standards\Squiz\Sniffs\WhiteSpace\ControlStructureSpacingSniff.SpacingAfterOpenBrace" were not
         found. You can remove them from "parameters: > skip:" section in your config.        
```

The false positives are not present if I run it with `--clear-cache`. But they are back as soon as remove the `--clear-cache`.

--- a/packages/EasyCodingStandard/packages/SniffRunner/src/File/File.php
+++ b/packages/EasyCodingStandard/packages/SniffRunner/src/File/File.php
@@ -5,6 +5,7 @@ namespace Symplify\EasyCodingStandard\SniffRunner\File;
 use PHP_CodeSniffer\Files\File as BaseFile;
 use PHP_CodeSniffer\Standards\Generic\Sniffs\CodeAnalysis\AssignmentInConditionSniff;
 use PHP_CodeSniffer\Standards\Squiz\Sniffs\PHP\CommentedOutCodeSniff;
+use SplFileInfo;
 use Symplify\EasyCodingStandard\Application\AppliedCheckersCollector;
 use Symplify\EasyCodingStandard\Error\ErrorAndDiffCollector;
 use Symplify\EasyCodingStandard\Skipper;
@@ -122,7 +123,9 @@ final class File extends BaseFile
     {
         $this->appliedCheckersCollector->addFileAndChecker($this->path, $this->resolveFullyQualifiedCode($code));
 
-        return ! $this->skipper->shouldSkipCodeAndFile($this->resolveFullyQualifiedCode($code), $this->path);
+        $absolutePath = (new SplFileInfo($this->path))->getRealPath();
+
+        return ! $this->skipper->shouldSkipCodeAndFile($this->resolveFullyQualifiedCode($code), $absolutePath);
     }
 
     /**
@@ -130,7 +133,9 @@ final class File extends BaseFile
      */
     public function addError($error, $stackPtr, $code, $data = [], $severity = 0, $fixable = false): bool
     {
-        if ($this->skipper->shouldSkipCodeAndFile($this->resolveFullyQualifiedCode($code), $this->path)) {
+        $absolutePath = (new SplFileInfo($this->path))->getRealPath();
+
+        if ($this->skipper->shouldSkipCodeAndFile($this->resolveFullyQualifiedCode($code), $absolutePath)) {
             return false;
         }
 

--- a/packages/EasyCodingStandard/src/Skipper.php
+++ b/packages/EasyCodingStandard/src/Skipper.php
@@ -101,7 +101,7 @@ final class Skipper
      */
     public function getUnusedSkipped(): array
     {
-        // global code skips always remove: ref https://github.com/Symplify/Symplify/issues/778
+        // always remove global code skips: ref https://github.com/Symplify/Symplify/issues/778
         foreach ($this->skippedCodes as $index => $value) {
             // skip all
             if ($value === null) {

--- a/packages/EasyCodingStandard/src/Skipper.php
+++ b/packages/EasyCodingStandard/src/Skipper.php
@@ -101,6 +101,14 @@ final class Skipper
      */
     public function getUnusedSkipped(): array
     {
+        // global code skips always remove: ref https://github.com/Symplify/Symplify/issues/778
+        foreach ($this->skippedCodes as $index => $value) {
+            // skip all
+            if ($value === null) {
+                unset($this->unusedSkipped[$index]);
+            }
+        }
+
         return $this->unusedSkipped;
     }
 

--- a/packages/EasyCodingStandard/tests/Skipper/SkipperRemoveUnusedTest.php
+++ b/packages/EasyCodingStandard/tests/Skipper/SkipperRemoveUnusedTest.php
@@ -29,7 +29,6 @@ final class SkipperRemoveUnusedTest extends TestCase
                 0 => 'someFile',
                 1 => '*/someDirectory/*',
             ],
-            'someSniff.someCode' => null,
             'someSniff.someOtherCode' => [
                 '*/someDirectory/*',
             ],
@@ -44,7 +43,6 @@ final class SkipperRemoveUnusedTest extends TestCase
             DeclareStrictTypesFixer::class => [
                 1 => '*/someDirectory/*',
             ],
-            'someSniff.someCode' => null,
             'someSniff.someOtherCode' => [
                 '*/someDirectory/*',
             ],

--- a/packages/EasyCodingStandardTester/README.md
+++ b/packages/EasyCodingStandardTester/README.md
@@ -68,7 +68,7 @@ final class YourFixerTest extends AbstractCheckerTestCase
 
 There is one extra method for sniff that doesn't fix the error, but only finds it:
 
-- `doTestWrongFile($wrongFile)` 
+- `doTestWrongFile($wrongFile)`
 
 ```php
 <?php declare(strict_types=1);
@@ -80,7 +80,7 @@ use Symplify\EasyCodingStandardTester\Testing\AbstractCheckerTestCase;
 final class YourSniffTest extends AbstractCheckerTestCase
 {
     // ...
-    
+
     public function testWrongCases(): void
     {
         $this->doTestWrongFile(__DIR__ . '/wrong/wrong.php.inc');

